src/tbp/monty/frameworks/models/goal_state_generation.py

```diff
@@ -981,6 +990,11 @@ def _check_conditions_for_hypothesis_test(self):
         if (
             len(pm_smaller_thresh) == 1 and (self.parent_lm.rng.uniform() <= 0.5)
         ) or len(pm_base_thresh) == 1:
+            # If we only have one object with a single hypothesis, we should not
+            # attempt to generate a goal state.
+            if len(self.parent_lm.get_evidence_for_object(top_id)) == 1:
```

Comment:
We check this in two places now (here and on line 967). Is this necessary? Would we ever get into this if branch down here if the first one is already true?